### PR TITLE
Update WebUI inline with v5.0.x

### PR DESCRIPTION
**Pull request by saddam213**
Update the WebUI to be inline with the latest LLamaSharp

- [x] Service for create/dispose of Models and Contexts
- [x] Consolidate Executor pages into single page

### CODE REVIEW

File: LLama/LLamaInstructExecutor.cs [hunk 1]
```diff
@@ -27,10 +28,11 @@ public class InstructExecutor : StatefulExecutorBase
         /// 
         /// </summary>
         /// <param name="context"></param>
+        /// <param name="logger"></param>
         /// <param name="instructionPrefix"></param>
         /// <param name="instructionSuffix"></param>
-        public InstructExecutor(LLamaContext context, string instructionPrefix = "\n\n### Instruction:\n\n",
-            string instructionSuffix = "\n\n### Response:\n\n") : base(context)
+        public InstructExecutor(LLamaContext context, ILogger logger = null!, string instructionPrefix = "\n\n### Instruction:\n\n",
```
Comment (martindevans) [inline LLama/LLamaInstructExecutor.cs:34]:
```suggestion
        public InstructExecutor(LLamaContext context, ILogger? logger = null, string instructionPrefix = "\n\n### Instruction:\n\n",
```

File: LLama/LLamaInteractExecutor.cs [hunk 2]
```diff
@@ -25,7 +26,8 @@ public class InteractiveExecutor : StatefulExecutorBase
         /// 
         /// </summary>
         /// <param name="context"></param>
-        public InteractiveExecutor(LLamaContext context) : base(context)
+        /// <param name="logger"></param>
+        public InteractiveExecutor(LLamaContext context, ILogger logger = null!) : base(context, logger)
```
Comment (martindevans) [inline LLama/LLamaInteractExecutor.cs:30]:
```suggestion
        public InteractiveExecutor(LLamaContext context, ILogger? logger = null) : base(context, logger)
```

File: LLama/LLamaStatelessExecutor.cs [hunk 3]
```diff
@@ -33,8 +35,10 @@ public class StatelessExecutor
         /// </summary>
         /// <param name="weights"></param>
         /// <param name="params"></param>
-        public StatelessExecutor(LLamaWeights weights, IContextParams @params)
+        /// <param name="logger"></param>
+        public StatelessExecutor(LLamaWeights weights, IContextParams @params, ILogger logger = null!)
```
Comment (martindevans) [inline LLama/LLamaStatelessExecutor.cs:39]:
```suggestion
        public StatelessExecutor(LLamaWeights weights, IContextParams @params, ILogger? logger = null)
```

File: LLama/LLamaInteractExecutor.cs (hunk 2 above)
Comment (martindevans) [reply LLama/LLamaInteractExecutor.cs:30]: You may need to change it to `ILogger?` in the base constructor as well

Comment (saddam213) [reply LLama/LLamaInteractExecutor.cs:30]: Ahh yeah sorry, I dont use that nullable nonsense, Ill change my VS settings for LLamaSharp so I dont keep doing it :p

File: LLama/LLamaWeights.cs [hunk 4]
```diff
@@ -80,10 +81,11 @@ public void Dispose()
         /// Create a llama_context using this model
         /// </summary>
         /// <param name="params"></param>
+        /// <param name="logger"></param>
         /// <returns></returns>
-        public LLamaContext CreateContext(IContextParams @params)
+        public LLamaContext CreateContext(IContextParams @params, ILogger logger = default!)
```
Comment (martindevans) [inline LLama/LLamaWeights.cs:86]:
```suggestion
        public LLamaContext CreateContext(IContextParams @params, ILogger? logger = default)
```

**Comment by saddam213** [comment]
Ill just focus on webUI, and stay out of the LLamaSharp project, will prolly speed everything up :)

**Comment by saddam213** [comment]
I don't think I need to spit the Params right now, there is no benefit to the WebUI to separate them currently.

I will also leave the WebAPI, I'm not sure what direction the original dev was working towards

LLamasStack has a full UI and WPF application for LLamaSharp anyway :)

**Pull request merged**